src/util/system.cpp

```diff
@@ -438,6 +438,13 @@ const fs::path& ArgsManager::GetDataDir(bool net_specific) const
     return path;
 }
 
+void ArgsManager::InitDefaultDataDir(const fs::path path)
+{
+    LOCK(cs_args);
+    if (m_init_default_datadir_path.empty())
```

Comment:
In commit "system: allow GUI to initialize default data dir" (7a9b786e00a5275ba012bc37a3796d1d35491161)

This `if` statment is unnecessary and causes `InitDefaultDataDir` to do nothing if it is called a second time. Would be saner to rename `InitDefaultDataDir` `SetDefaultDataDir` and just set the path unconditionally without unnecessary stickiness and no way of knowing whether the path was set.